test_flows for ap-metrics-response as part of the refactoring

RanRegev opened the pull request:
added first few lines:
confing request and checking for ACK
the test fails at the moment

abelog committed:
tests: add device to boardfarm config

Add "prplWRT" device "rax40-1" to boardfarm config.

In order to this config to work, you have to:
1. Create docker bridge network with name "prplMesh-net-rax40-1".
`docker network create --driver=bridge prplMesh-net-rax40-1`
2. Include interface connected to the board in docker network.
`ip link set $RAX40_IFACE master $DOCKER_BRIDGE`
3. Config relies on RAX40 consoles to be at "/dev/ttyUSB0". You may have
to adjust this path to fit your system.

Those steps are required, as regular user is not permitted to attach
interfaces to the bridge.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>

abelog committed:
tests: boardfarm: add prplWRT device

Add prplWRT class which represents device running prplWRT with prplMesh
installed. This class has required methods to init the board and execute
InitialApConfig test on board.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>

abelog committed:
tests:  environment.py: complete prplWRT entities

prplWRT entities are not debugged, as there was no way to test them at
the moment of commit.

`_device_wait_for_log` method: count in start_line given to it, format
return as it's expected by callers.
`ALEntityPrplWrt` constructor: get main bridge name from config. Parse
output of console in Python to be less dependent on board.
`RadioHostapd` constructor: remove usage of `ip -o` as `-o` option is
bugged on RAX40 and mutes output of `ip` command.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>

abelog committed:
tests: boardfarm: InitialApConfig - too greedy regex search

Currently, test is trying to match r".* Controller configuration \(WSC
M2 Encrypted Settings\)") which is greedy and captures all chars before
key message. Chars captured by `.*` are not important for us.

Get rid of `.*` in the beginning of regex. Reorder expected logs to
match them in right order.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>

abelog committed:
tests: boardfarm: move to ssh as primary connection

Currently, boardfarm tests are using UART (serial) connection as main
channel to interact with the board. Sadly, but UART on RAX40 board
frequently suffers from tty overruns. Hence, tests are frequently failed
due to corrupted input line.

Moving to ssh as primary channel resolves issue above. It's faster, so
chance of meeting input overruns is significantly low. But, via ssh we
are unable to see logs from the Intel WiFi driver. Logs from the driver are
used to detect when device has prplMesh in operational state. As those
logs are not longer shown, we have to come up with other way to check
when prplMesh is operational.

Add fallback mechanism: if IP address of DUTs wan is not set - retrieve
IP subnet from docker network of set up and choose .245 address. This
configuration happens over UART.
Move from UART to SSH as primary communication channel.
Add prplmesh_status_check method to tests, to check if prplMesh on
device is operational.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>

Commit:
agent: backhaul_manager: Refactor handler for AP_METRICS_RESPONSE_MESSAGE

Create new class ExpectedApMetricsResponse for
easy handling of the logic of AP Metrics query and response.
The class saves the expected bssids responses and
collect them together before sending the combined
response to the controller

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>
Signed-off-by: Ran Regev <ran.regev@devalore.com>

abelog committed:
tests: boardfarm: docker devices support of InitialAPConfig teardown

Teardown section of InitialAPConfig test introduced by port of this test
to prplWRT devices is incompatible with dockerized devices. It uses
`device.send` method which is not available for docker devices.
Also, prplmesh_status_check method isn't working for docker devices due
to missing link between AlEntityDocker and PrplMeshDocker device.

Following EAFP strategy, surround sent of Ctrl+C in try-catch block.
Add link between AlEntityDocker and PrplMeshDocker as it done for
prplWRT device and restore prplmesh_status_check method functionality.

Signed-off-by: Anton Bilohai <a.bilohai@inango-systems.com>

mariomaz committed:
tlvf: fix tlv_mac_tlv in msgTopologyDiscovery

Change the type of field tlv_mac_tlv from tlvAlMacAddressType to
tlvMacAddress.

Note that this change does not affect to autogenerated files nor to
source code.

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

mariomaz committed:
tlvf: rename tlvAlMacAddressType to tlvAlMacAddress

No TLV type ends with prefix 'Type' except this one. Rename it to look
like the rest of TLVs

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

itayx committed:
common: prplmesh_utils.sh: add bml execute func

This is a preparative commit for "common: prplmesh_utils.sh: use BML for
operational check"

Add execute BML function

Signed-off-by: itay elenzweig <itayx.elenzweig@intel.com>

itayx committed:
common: prplmesh_utils.sh: use BML operational check

prplmesh_utils.sh status returns an operational status determined by
parsing the logs. In some platforms logs wrap around could return "fail"
status although system is operational state.

To resolve this issue, the status check should use the BML command
bml_get_device_operational_radios which returns the status of the
current device using the bridge's MAC address.

Use BML to check operational status

Signed-off-by: itay elenzweig <itayx.elenzweig@intel.com>

Commit:
devcontainer: add missing packages

Latest changes introduced dependecies to new tools
that are not currently installed in the devcontainer.
Install the missing tools as part of the creation of the devcontainer.

Signed-off-by: Vitaly Bukhovsky <vitaly@devalore.com>

Commit:
vscode_launch: beerocks_fronthaul support

Change beerocks_monitor attach tasks to beerocks_fronthaul

Signed-off-by: Vitaly Bukhovsky <vitaly@devalore.com>

mariomaz committed:
bcl: change return type of linux_get_iface_index

Method linux_get_iface_index() is a wrapper for if_nametoindex but
return types differ.

Change return type of linux_get_iface_index from int to uint32_t and
also write a message to log in case of error.

This is a preparatory commit. This method is currently not used but it
will be in next commits.

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

mariomaz committed:
btl: add iface_name param to send_cmdu functions

Add new iface_name parameter to send_cmdu() functions to specify on
which interface the transport should send given CMDU.

This is a preparatory commit to later implement neighbor multicast of
Topology Discovery message.

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

mariomaz committed:
agent: backhaul: send Topology Discovery on all interfaces

Currently, we sent a single Topology Discovery message through the
bridge.

However, according to IEEE1905.1, the message should include a MAC
Address TLV which contains the address of the interface on which you
send the message. Thus, a different message should be sent on each
interface.

Therefore, the agent should send a separate Discovery message on each
interface.

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

Commit:
documentation: agent: Add UML diaram for Backhaul STA steering

According to the test plan and source code
create the UML diagram which describes
data flow and methods for Backhaul STA steering.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
framework: tlvf: Turn on Backhaul STA Steering TLV's

For auto-generating TLV for Backhaul STA Steering
Request/Response needs to add yaml files to the tlv_conf.yaml.
It is turning on auto-generation code from those TLV.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

Commit:
tlvf: CmduMessageRx: Add case for Backhaul STA Steering

In the method parseNextTlv() missing cases for
Backhaul STA Steering Request/Response TLV's.

Add missing cases with correct values for
message types.

Signed-off-by: Vladyslav Tupikin <v.tupikin@inango-systems.com>

mariomaz committed:
btl: add if_index to UDS header of received message

Modify transport to include which interface the message was received on.

Interface index will later be stored in topology database whenever a
Topology Discovery message is received.

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

mariomaz committed:
agent: backhaul: extend topology database with if_index and MAC

When processing Topology Discovery message we must store the IEEE
1905.1 AL MAC address of the transmitting device together with the MAC
address of the interface on which the message is transmitted and the
interface on which such message is received. Currently we are storing
AL MAC address only.

This additional information is required to build Topology Response and
Link Metric Response messages properly.

Signed-off-by: Mario Maz <mmazzapater@gmail.com>

Commit:
btlvf: format yaml files

Signed-off-by: Vitaly Bukhovsky <vitaly@devalore.com>

RanRegev committed:
test: test_flowys.py added test_ap_metrics_interval_response

added test for ap metrics query/response
tests that there is exactly one response from
each agent within the time of the interval

MAP-4.7.4_ETH_FH24G:netgear-rax40
MAP-4.7.4_ETH_FH5GL:netgear-rax40
MAP-4.7.4_ETH_FH5GH:netgear-rax40
MAP-4.7.5_ETH_FH24G:netgear-rax40
MAP-4.7.5_ETH_FH5GL:netgear-rax40
MAP-4.7.5_ETH_FH5GH:netgear-rax40
MAP-4.7.6_ETH_FH24G:netgear-rax40
MAP-4.7.6_ETH_FH5GL:netgear-rax40
MAP-4.7.6_ETH_FH5GH:netgear-rax40

rmelotte committed:
bwl: initialize SStaStats

SStaStats values were not initialized, which could have lead to
unexpected results if they are read before being assigned to.

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

rmelotte committed:
mon_wlan_hal_dwpal: parse SNR and ShortTermRSSIAverage as strings

The previous parsing of SNR and ShortTermRSSIAverage was failing,
leading to a segmentation fault of beerocks_fronthaul.

Parse them as strings instead. Since they contain space-separated
values, split them later.

Fixes https://github.com/prplfoundation/prplMesh/issues/1438

MAP-4.3.2_ETH_FH24G:netgear-rax40

Signed-off-by: Raphaël Mélotte <raphael.melotte@mind.be>

arnout commented on agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp:
```diff
@@ -2201,6 +2208,7 @@ bool backhaul_manager::handle_1905_1_message(ieee1905_1::CmduMessageRx &cmdu_rx,
     }
     case ieee1905_1::eMessageType::MULTI_AP_POLICY_CONFIG_REQUEST_MESSAGE: {
         return handle_multi_ap_policy_config_request(cmdu_rx, src_mac);
+    }
```
Could you make a small PR with just this fix?

RanRegev committed:
slave: backhaul manager: fixed weird curly brackets

fixed a strange code in handle_1905_1_message's switch case

Signed-off-by: Ran Regev <ran.regev@devalore.com>

RanRegev commented:
here:
https://github.com/prplfoundation/prplMesh/pull/1451

RanRegev commented:
github.com/prplfoundation/prplMesh/pull/1451

RanRegev commented:
https://github.com/prplfoundation/prplMesh/pull/1451


On Thu, Jun 18, 2020 at 1:46 PM Ran Regev <ran.regev@devalore.com> wrote:

>    github.com/prplfoundation/prplMesh/pull/1451
>
>

Commit:
prplmesh_uttils.sh: use SIGTERM in killall_program()

Currently prplmesh_utills.sh stop sends SIGKILL to all processes.
Since SIGKILL cannot be handled monitor/agent/controller
processes are not shutdown properly.
As a result old wpa ctrl socket are not detached and files
are not deleted.

Change killall_program() to send SIGTERM instead on default.

Signed-off-by: Alex Kanter <alexx.kanter@intel.com>

arnout committed:
mergify: allow status check failures

We have a requirement for mergify that no status checks should fail.
This is kind of redundant, because mergify will only do the merge if no
*required* status checks fail anyway.

Since we now have a non-required status check (UTAF) that often fails,
we really need to remove that redundant requirement.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

RanRegev commented:
This PR is too messy.
Creating a new PR with only two commits.
#1467
